src/marshmallow/validate.py

```diff
@@ -32,6 +32,60 @@ def _repr_args(self) -> str:
         """
         return ""
 
+    def __call__(self, value):
```

Comment:
What about adding a type annotation using generic? All (must part?) validators return the same type that passed for validation.
```python
T = TypeVar("T")

class Validator(ABC):

    @abstractmethod
    def __call__(self, value: T) -> T:
        pass
```
This will help us infer types and validate them through static analysis.